src/main/java/com/shrralis/ssdemo1/controller/AuthRestController.java

```diff
@@ -12,96 +12,64 @@
-import org.springframework.security.core.Authentication;
-import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-
-import static java.util.Map.entry;
+import javax.mail.MessagingException;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
 
 /**
  * @author shrralis (https://t.me/Shrralis)
  * @version 1.0
  * Created 12/20/17 at 5:59 PM
  */
 @RestController
-@RequestMapping("auth")
+@RequestMapping("/auth")
 public class AuthRestController {
-	private static final Logger logger = LoggerFactory.getLogger(AuthRestController.class);
-	@Autowired
-	IAuthService service;
-    @Autowired
-    private AuthenticationTrustResolver authenticationTrustResolver;
-
-	@RequestMapping({ "/login", "/logout" })
-	public JsonResponse checkCurrSession(@RequestParam(value = "error", required = false) Boolean isError) {
-		if (isError != null) {
-			if (isError) {
-				return new JsonResponse(JsonError.Error.UNEXPECTED);
-			} else {
-				return new JsonResponse(JsonError.Error.NO_ERROR);
-			}
-		}
-
-		AuthorizedUser authorizedUser = AuthorizedUser.getCurrent();
 
-		if (isCurrentAuthenticationAnonymous() || authorizedUser == null) {
-			return new JsonResponse(Map.ofEntries(entry("logged_in", false)));
-		}
+	private IAuthService service;
+	private AuthenticationTrustResolver authTrustResolver;
 
-
-		return new JsonResponse(Map.ofEntries(
-				entry("id", authorizedUser.getId()),
-				entry("login", authorizedUser.getUsername()),
-				entry("type", authorizedUser.getType())
-		));
+	@Autowired
+	public AuthRestController(IAuthService service,
+	                          AuthenticationTrustResolver authTrustResolver) {
+		this.service = service;
+		this.authTrustResolver = authTrustResolver;
 	}
 
-    @RequestMapping(
-            value = "signUp",
-		    consumes = MediaType.APPLICATION_JSON_UTF8_VALUE,
-		    produces = MediaType.APPLICATION_JSON_UTF8_VALUE,
-		    method = RequestMethod.POST)
-    public JsonResponse signUp(@RequestBody RegisterUserDTO user) throws AbstractShrralisException {
-        return service.signUp(user);
-    }
-
-    private boolean isCurrentAuthenticationAnonymous() {
-        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+	@GetMapping(value = "/login", params = "error")
+	public JsonResponse showErrorOnLogin() {
+//		TODO: finish this
+		return null;
+	}
 
-        return authenticationTrustResolver.isAnonymous(authentication);
-    }
+	@GetMapping({ "/login", "/logout" })
+	public JsonResponse getCurrSession() {
+		return new JsonResponse(service.getCurrentSession(SecurityContextHolder.getContext().getAuthentication(),
```

Comment:
You may create static method in service that will hide work with SecurityContextHolder